liferay-echo/liferay-portal
# LPS-169416 New User Segment sidebar trigger switch is not correctly controlled

Motivation
=======
The motivation behind this ticket is to solve an accessibility issue related to the Segments Editor. Is about the toggle button which is used to change between the edit and view mode of the Segments Editor and opens/closes the Segments Contributors Panel. This button isn't accessible since it doesn't inform the user with the proper informative messages and the button is not related to the panel it opens.
[LPS-169416](https://issues.liferay.com/browse/LPS-169416)

Proposed Solution
========
To solve the problem we change the button icon and add the labels, titles and roles necessary. Also we use the focus method combined with tabIndex=-1 attribute to handle the focus when the panel is opened. 

Steps to Verify
========
1. Go to People -> Segments
2. Create a Segments by clicking on (+)
3. Activate ScreenReader and navigate by the keyboard to the configuration button
4. Click enter on the button and check that the focus is changed to the opened panel
5. Listen to the ScreenReader and check that informs of the two possible messages of the button (Enter View/Edit Mode) and says the panel name (Segments Contributors Panel) correctly

<img width="1792" alt="Screenshot 2023-03-28 at 19 11 08" src="https://user-images.githubusercontent.com/91207948/228316615-b50f2f65-b4b7-4e09-b3b3-961598180e5c.png">
